src/Microsoft.AspNetCore.Mvc.ViewFeatures/Internal/PagedBufferedTextWriter.cs

```diff
@@ -83,11 +83,19 @@ public override void Write(char value)
 
         public override void Write(char[] buffer)
         {
-            Write(buffer, 0, buffer.Length);
+            if (buffer != null)
+            {
+                Write(buffer, 0, buffer.Length);
+            }
         }
 
         public override void Write(char[] buffer, int index, int count)
         {
+            if (buffer == null)
+            {
+                throw new ArgumentNullException(nameof(buffer), Resources.ArgumentNull_Buffer);
```

Comment:
Get rid of the resource and just throw `ArgumentNullException(nameof(buffer))`. We fortunately don't use this constructor overload anywhere in MVC.